src/lib.rs

```diff
@@ -191,82 +194,92 @@ impl core::fmt::Display for PReg {
     }
 
     /// Returns whether the given register is part of the set.
     pub fn contains(&self, reg: PReg) -> bool {
-        debug_assert!(reg.index() < 256);
-        let bit = reg.index() & 127;
-        let index = reg.index() >> 7;
-        self.bits[index] & (1u128 << bit) != 0
+        let (index, bit) = Self::split_index(reg);
+        self.bits[index] & (1 << bit) != 0
     }
 
     /// Add a physical register (PReg) to the set, returning the new value.
     pub const fn with(self, reg: PReg) -> Self {
-        debug_assert!(reg.index() < 256);
-        let bit = reg.index() & 127;
-        let index = reg.index() >> 7;
+        let (index, bit) = Self::split_index(reg);
         let mut out = self;
-        out.bits[index] |= 1u128 << bit;
+        out.bits[index] |= 1 << bit;
         out
     }
 
     /// Add a physical register (PReg) to the set.
     pub fn add(&mut self, reg: PReg) {
-        debug_assert!(reg.index() < 256);
-        let bit = reg.index() & 127;
-        let index = reg.index() >> 7;
-        self.bits[index] |= 1u128 << bit;
+        let (index, bit) = Self::split_index(reg);
+        self.bits[index] |= 1 << bit;
     }
 
     /// Remove a physical register (PReg) from the set.
     pub fn remove(&mut self, reg: PReg) {
-        debug_assert!(reg.index() < 256);
-        let bit = reg.index() & 127;
-        let index = reg.index() >> 7;
-        self.bits[index] &= !(1u128 << bit);
+        let (index, bit) = Self::split_index(reg);
+        self.bits[index] &= !(1 << bit);
     }
 
     /// Add all of the registers in one set to this one, mutating in
     /// place.
     pub fn union_from(&mut self, other: PRegSet) {
-        self.bits[0] |= other.bits[0];
-        self.bits[1] |= other.bits[1];
+        for i in 0..self.bits.len() {
+            self.bits[i] |= other.bits[i];
+        }
     }
 }
 
 impl IntoIterator for PRegSet {
     type Item = PReg;
     type IntoIter = PRegSetIter;
     fn into_iter(self) -> PRegSetIter {
-        PRegSetIter { bits: self.bits }
+        PRegSetIter {
+            bits: self.bits,
+            cur: 0,
+        }
     }
 }
 
 pub struct PRegSetIter {
-    bits: [u128; 2],
+    bits: [Bits; PRegSet::LEN],
+    cur: usize,
 }
 
 impl Iterator for PRegSetIter {
     type Item = PReg;
     fn next(&mut self) -> Option<PReg> {
-        if self.bits[0] != 0 {
-            let index = self.bits[0].trailing_zeros();
-            self.bits[0] &= !(1u128 << index);
-            Some(PReg::from_index(index as usize))
-        } else if self.bits[1] != 0 {
-            let index = self.bits[1].trailing_zeros();
-            self.bits[1] &= !(1u128 << index);
-            Some(PReg::from_index(index as usize + 128))
-        } else {
-            None
+        loop {
+            let bits = self.bits.get_mut(self.cur)?;
+            if *bits != 0 {
+                let bit = bits.trailing_zeros();
+                *bits ^= 1 << bit;
```

Comment:
Can we stick with the `&= !(1 << bit)` form we had before, here? Equivalent in this case and not a huge deal either way, just a preference, but this form for me is slightly more clear in intent (clearing a bit, vs. XOR which I ordinarily see for flipping the bit).